### PR TITLE
Use treemap for extra data maps if map had only 1 entry

### PR DESCRIPTION
**Pull request by HenrikJannsen**
Replacing HashMap with TreeMap for maps which only contains 1 entry is safe as order is irrelevant in that case.

Those affected maps had never more then 1 entry in the past.

## Summary by CodeRabbit

* **Refactor**
  * Extra-data maps now use a deterministic, ordered representation to ensure consistent serialization across components.

* **Bug Fixes**
  * Incoming oversized/invalid extra-data is sanitized and triggers warnings to prevent malformed payloads.

* **Tests**
  * Added comprehensive protobuf round-trip and edge-case tests for extra-data handling and serialization determinism.

* **Chores**
  * Removed an unused validator import and clarified related error messages.

### CODE REVIEW

File: core/src/test/java/bisq/core/dao/state/model/governance/CompensationProposalTest.java [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+        hashMap.put(CompensationProposal.BURNING_MAN_RECEIVER_ADDRESS, "receiverAddress");
+        protobuf.Proposal hashMapProto = protobuf.Proposal.newBuilder()
+                .setName("name")
+                .setLink("link")
+                .setVersion(1)
+                .setCreationDate(1)
+                .setTxId("txId")
+                .setCompensationProposal(protobuf.CompensationProposal.newBuilder()
+                        .setRequestedBsq(10_000)
+                        .setBsqAddress("bsqAddress"))
```
[the remark this answers is not in the document]
Comment (HenrikJannsen) [reply core/src/test/java/bisq/core/dao/state/model/governance/CompensationProposalTest.java:70]: This is wrong. You can pass a nested builder in protobuf.

File: p2p/src/test/java/bisq/network/p2p/storage/payload/MailboxStoragePayloadTest.java [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+        assertTrue(payload.getExtraDataMap() instanceof TreeMap);
+        assertEquals(CUSTOM_TTL, payload.getTTL());
+        assertArrayEquals(hashMapProto.toByteArray(), payload.toProtoMessage().toByteArray());
+    }
+
+    @Test
+    public void fromProtoPreservesNullExtraDataMap() throws NoSuchAlgorithmException {
+        protobuf.MailboxStoragePayload proto = newMailboxStoragePayload(MailboxStoragePayload.TTL)
+                .toProtoMessage()
+                .getMailboxStoragePayload();
```
[the remark this answers is not in the document]
Comment (HenrikJannsen) [reply p2p/src/test/java/bisq/network/p2p/storage/payload/MailboxStoragePayloadTest.java:78]: This is wrong. You can pass a nested builder in protobuf.

File: core/src/main/java/bisq/core/trade/statistics/TradeStatistics3.java [hunk 3]
```diff
@@ -231,13 +231,14 @@ enum PaymentMethodMapper {
...
     private final byte[] hash;
+
     // Should be only used in emergency case if we need to add data but do not want to break backward compatibility
     // at the P2P network storage checks. The hash of the object will be used to verify if the data is valid. Any new
     // field in a class would break that hash and therefore break the storage mechanism.
     @Nullable
     @JsonExclude
     @Getter
-    private final Map<String, String> extraDataMap;
+    private final TreeMap<String, String> extraDataMap;
```
[the remark this answers is not in the document]
Comment (HenrikJannsen) [reply core/src/main/java/bisq/core/trade/statistics/TradeStatistics3.java:241]: This is intentional as TreeMap is required for canonical order when data gets serialized.
